### PR TITLE
backend/feat/scheduled-trips-for-route

### PR DESCRIPTION
**Pull request by yashikakaushik**
## Type of Change

- [X] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [X] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/spec/API/ScheduledTrips.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  TrackingResp:
+    vehicleTrackingInfo : "[VehicleInfo]"
+  VehicleInfo:
+    location: LatLong
+    nextStop: RouteStopMapping
+    vehicleId: Text
+
+apis:
+  - GET:
+      endpoint: /scheduled/{routeId}/trips
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/API/ScheduledTrips.yaml:19]: `/track/{routeCode}`

File: Backend/app/rider-platform/rider-app/Main/spec/API/ScheduledTrips.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      params:
+        routeId: Id Route
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/API/ScheduledTrips.yaml:21]: `routeCode: Text`

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/ScheduledTrips.hs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+mkLatLong lat_ lon_ =
+  Maps.LatLong
+    { lat = lat_,
+      lon = lon_
+    }
+
+pairWithNext :: [a] -> [(a, a)]
+pairWithNext xs = zip xs (List.tail xs)
+
+data VehicleInfoForRoute = VehicleInfoForRoute
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/ScheduledTrips.hs:95]: Can it all be Maybe ?

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/ScheduledTrips.hs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  let sortedStops = sortBy (compare `on` RouteStopMapping.sequenceNum) reqStops
+      stopPairs = map (\(x, y) -> (x, y)) (pairWithNext sortedStops)
+
+  trackingResp <-
+    forM routeInfoFromRedis $ \singleRouteInfo -> do
+      let vehicleId_ = fst singleRouteInfo
+      let vehicleInfo = snd singleRouteInfo
+
+      minDistancesWithPoint <-
+        forM stopPairs $ \stopPair -> do
```
Comment (piyushKumar-1) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/ScheduledTrips.hs:60]: can you do this outside the above loop ?

**Review by piyushKumar-1 — APPROVED** [review]

**Pull request merged**